src/test/run-make/issue-11908/Makefile

```diff
@@ -12,12 +12,11 @@ all:
 	$(RUSTC) foo.rs --crate-type=dylib
 	mv $(call DYLIB,foo) $(TMPDIR)/other
 	$(RUSTC) foo.rs --crate-type=dylib
-	$(RUSTC) bar.rs -L $(TMPDIR)/other 2>&1 | \
-		grep "multiple dylib candidates"
+	# dylibs fail with strange errors, but not assertions.
+	! $(RUSTC) bar.rs -L $(TMPDIR)/other 2>&1 | grep "assertion failed"
```

Comment:
This is really worrisome to me, this should generate the same error message it seems like? Either that or the two dylibs here need to have different hashes to generate the same error.